bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/SonnenHandler.java

```diff
@@ -0,0 +1,278 @@
+import javax.measure.quantity.Power;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.sonnen.communication.SonnenJSONCommunication;
+import org.openhab.binding.sonnen.communication.SonnenJsonDataDTO;
+import org.openhab.binding.sonnen.utilities.Helper;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SonnenHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Christian Feininger - Initial contribution
+ */
+@NonNullByDefault
+public class SonnenHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SonnenHandler.class);
+
+    private SonnenConfiguration config = new SonnenConfiguration();
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private SonnenJSONCommunication serviceCommunication;
+
+    private boolean automaticRefreshing = false;
+
+    private Map<String, Boolean> linkedChannels = new HashMap<>();
+
+    public SonnenHandler(Thing thing) {
+        super(thing);
+        serviceCommunication = new SonnenJSONCommunication();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing sonnen handler for thing {}", getThing().getUID());
+        config = getConfigAs(SonnenConfiguration.class);
+        boolean validConfig = true;
+        String statusDescr = null;
+        if (config.refreshInterval < 0 && config.refreshInterval > 1000) {
+            statusDescr = "Parameter 'refresh Rate' greater then 0 and less then 1001.";
+            validConfig = false;
+        }
+        if (config.hostIP == null) {
+            statusDescr = "IP Address must be configured!";
+            validConfig = false;
+        }
+
+        Helper message = new Helper();
+        message.setStatusDescription(statusDescr);
+        if (validConfig) {
+            serviceCommunication.setConfig(config);
+            scheduler.submit(() -> {
+                if (serviceCommunication.refreshBatteryConnection(message, this.getThing().getUID().toString())) {
+                    updateStatus(ThingStatus.ONLINE);
+                    updateLinkedChannels();
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            message.getStatusDesription());
+                }
```

Comment:
Please confirm but I think all this block could be replaced by:
```
if (updatebatteryData()) {
    updateLinkedChannels();
}
```

Reply:
No, I am wrong, your are setting a different OFFLINE thing status.

Reply:
Indeed, you are right. Didn't see this.